### PR TITLE
implement failures suppressing for @Flaky tests

### PR DESCRIPTION
**Pull request by int02h**

### CODE REVIEW

File: subprojects/android-test/test-annotations/src/main/kotlin/com/avito/android/test/annotations/Flaky.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.avito.android.test.annotations
+
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
+annotation class Flaky(val reason: String = NO_REASON)
```
Comment (dsvoronin) [inline subprojects/android-test/test-annotations/src/main/kotlin/com/avito/android/test/annotations/Flaky.kt:10]: i want to see link to "Dealing with flaky tests" page in doc here
even if it's a stub page for now

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt [hunk 2]
```diff
@@ -87,6 +87,10 @@ abstract class InstrumentationTestsTask @Inject constructor(
     @Optional
     val suppressFailure = objects.property<Boolean>()
 
+    @Input
+    @Optional
```
Comment (dsvoronin) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt:91]: why it is optional? we can always use default here

File: subprojects/common/report-viewer/src/main/kotlin/com/avito/report/ReportsAddApiImpl.kt [hunk 3]
```diff
@@ -200,6 +201,16 @@ internal class ReportsAddApiImpl(private val requestProvider: JsonRpcRequestProv
             reportData["build_id_set"] = mapOf("\$fillSet" to listOf(buildId))
         }
 
+        when (val flakiness = test.flakiness) {
```
Comment (dsvoronin) [inline subprojects/common/report-viewer/src/main/kotlin/com/avito/report/ReportsAddApiImpl.kt:204]: please add tests (same as for other params)

File: subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/UiTestCheck.kt [hunk 4]
```diff
@@ -15,6 +15,11 @@ open class UiTestCheck(context: String) : SuppressibleBuildStep(context),
 
     var configurations = mutableListOf<String>()
 
+    /**
+     * Whether skip results of tests marked with @Flaky or not
+     */
+    var suppressFlaky: Boolean = false
```
Comment (dsvoronin) [inline subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/UiTestCheck.kt]: FYI: could be an interface like `impactAnalysisAware`

**Review by dsvoronin — CHANGES_REQUESTED** [review]

File: subprojects/android-test/test-annotations/src/main/kotlin/com/avito/android/test/annotations/Flaky.kt (hunk 1 above)
Comment (int02h) [reply subprojects/android-test/test-annotations/src/main/kotlin/com/avito/android/test/annotations/Flaky.kt:10]: Docs added

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt (hunk 2 above)
Comment (int02h) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt:91]: Discussed. Right now we won't change the behavior of the optional properties of this task.

File: subprojects/common/report-viewer/src/main/kotlin/com/avito/report/ReportsAddApiImpl.kt (hunk 3 above)
Comment (int02h) [reply subprojects/common/report-viewer/src/main/kotlin/com/avito/report/ReportsAddApiImpl.kt:204]: Done

File: subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/UiTestCheck.kt (hunk 4 above)
Comment (int02h) [reply subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/UiTestCheck.kt]: Done. Create an interface `FlakyAwareBuildStep`.

**Review by dsvoronin — APPROVED** [review]

**Pull request merged**